python/src/trezorlib/cli/firmware.py

```diff
@@ -607,43 +650,68 @@ def update(
     against downloaded firmware fingerprint. Otherwise fingerprint is checked
     against data.trezor.io information, if available.
     """
-    if sum(bool(x) for x in (filename, url, version)) > 1:
-        click.echo("You can use only one of: filename, url, version.")
-        sys.exit(1)
+    with obj.client_context() as client:
+        if sum(bool(x) for x in (filename, url, version)) > 1:
+            click.echo("You can use only one of: filename, url, version.")
+            sys.exit(1)
 
-    if not dry_run and not client.features.bootloader_mode:
-        click.echo("Please switch your device to bootloader mode.")
-        sys.exit(1)
+        if filename:
+            firmware_data = filename.read()
+        else:
+            if not url:
+                url, fp = find_best_firmware_version(
+                    client=client, version=version, beta=beta, bitcoin_only=bitcoin_only
+                )
+                if not fingerprint:
+                    fingerprint = fp
+
+            firmware_data = download_firmware_data(url)
+
+        if not raw and not skip_check:
+            validate_firmware(
+                firmware_data=firmware_data,
+                fingerprint=fingerprint,
+                bootloader_onev2=_is_bootloader_onev2(client),
+                model=client.model,
+            )
 
-    if filename:
-        firmware_data = filename.read()
-    else:
-        if not url:
-            url, fp = find_best_firmware_version(
-                client=client, version=version, beta=beta, bitcoin_only=bitcoin_only
+            if not raw:
+                firmware_data = extract_embedded_fw(
+                    firmware_data=firmware_data,
+                    bootloader_onev2=_is_bootloader_onev2(client),
+                )
+
+        if dry_run:
+            click.echo("Dry run. Not uploading firmware to device.")
+            return
+
+        strict_upgrade = _is_strict_update(client, firmware_data)
+        ilu_supported = _is_ilu_supported(client)
+
+        if not client.features.bootloader_mode and ilu_supported and strict_upgrade:
```

Comment:
just this:
```suggestion
        if not client.features.bootloader_mode:
```

Reply:
I've made a change so that trezorctl fw update always switches to the bootloader, using INSTALL_UPGRADE if the firmware is newer, or using STOP_AND_WAIT in other cases. 3139f02150230bdfd0bddd881ce1c7cfdd67e406